pkg/util/cmd/encoder.go

```diff
@@ -327,11 +329,10 @@ func (s *tidbEncoder) closeOpenNamespaces() {
 
 /* array encoder part */
 func (s *tidbEncoder) addElementSeparator() {
-	length := s.line.Len()
-	if length == 0 {
+	if s.line.Len() <= 0 {
 		return
 	}
-	switch s.line.Bytes()[length-1] {
+	switch s.line.Bytes()[s.line.Len()-1] {
```

Comment:
I don't think this makes any difference. If it's not thread-safe, it will still panic.

Reply:
Encoder is not thread-safe, same for pingcap/log or zap. I'll guess the root cause is `EncodeEntry`. The modifications here may not make any difference. I just change it for changes.

Reply:
It is reverted.